code/modules/mining/mine_items.dm

```diff
@@ -709,6 +709,57 @@ proc/move_mining_shuttle()
 	if(!loaded)
 		to_chat(user, "<span class='info'>\The [src] is empty.</span>")
 
+/obj/item/weapon/lazarus_injector/advanced
+	name = "Advanced lazarus injector"
+	desc = "A lazarus injector further enhanced with a nanomachine solution. Allows for the complete regeneration of lesser beings"
+	icon_state = "adv_lazarus_hypo"
+
+/obj/item/weapon/lazarus_injector/advanced/update_icon()
+	..()
+	if(loaded)
+		icon_state = "adv_lazarus_hypo"
+	else
+		icon_state = "adv_lazarus_empty"
+
+/obj/item/weapon/lazarus_injector/advanced/afterattack(atom/target, mob/user, proximity_flag)
```

Comment:
you mean like this? because if so, it doesn't refresh the mob drops at all
http://pastebin.com/0nztiHDQ